### PR TITLE
popup example

### PR DESCRIPTION
**Pull request by peterqliu**
redo of https://github.com/mapbox/mapbox-gl-js/pull/1008

cc/ @jfirebaugh

### CODE REVIEW

File: docs/_posts/examples/3400-01-08-html-popup.html [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  var markers = new mapboxgl.GeoJSONSource({ data: geoJSON });
+  map.addSource('markers', markers);
+
+  //set a click listener on the map object, to produce a popup whenever a click registers near a marker
+  map.on('click', function(e) {
+      map.featuresAt(e.point, {radius: 15}, function(err, features) {
+          if (err) throw err;
+          if (features[0]) {
+          	var markerPosition = features[0].properties.coordinates;
+          	var popupText = '<strong>'+features[0].properties.title+'</strong><br>'+ features[0].properties.address1+'<br>'+features[0].properties.address2;
```
Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-08-html-popup.html]: We should set a good example here and correctly HTML-escape the interpolated properties.

**Comment by jfirebaugh** [comment]
This example demonstrates very nicely the need for https://github.com/mapbox/mapbox-gl-js/issues/1002 and https://github.com/mapbox/mapbox-gl-js/issues/1010 -- I'm going to hold on merging until those are implemented (started work on them today).

File: docs/_posts/examples/3400-01-08-html-popup.html (hunk 1 above)
Comment (tmcw) [inline docs/_posts/examples/3400-01-08-html-popup.html]: PS: saw the other day that [bevacqua wrote a new html sanitizer](https://github.com/bevacqua/insane) that's way smaller than `sanitize-caja`, if we need to eat that complexity again.

Comment (jfirebaugh) [inline docs/_posts/examples/3400-01-08-html-popup.html]: I'm hoping not to have to. I gave Popup both `setText` and `setHTML` methods, with the expectation that the common use case will be VT data + BYO templating library.

Comment (peterqliu) [inline docs/_posts/examples/3400-01-08-html-popup.html]: Should I be doing the escape here in the example, or would it be incorporated as part of `.setHTML()` ? All the readings I've done suggest escaping via `element.appendChild(document.createTextNode(unsafe_str))`, but that seems like a duplicate of `.setHTML()`

**Comment by jfirebaugh** [comment]
Overtaken by a0606cc11df767df59ef64652c4d127cd8a10d59.

**Pull request closed without merging**